nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryQuery.java

```diff
@@ -3323,7 +3382,11 @@ public void testQueryIdListNotFromUuidButFromArrayMentionedInWhereClause() throw
         String nxql = "SELECT dc:subjects/*1 FROM File WHERE ecm:name = 'doc1' AND dc:subjects/*1 IS NULL";
         DocumentModelList dml = session.query(nxql, null, 10, 0, true);
         assertEquals(0, dml.size()); // null docs filtered out
-        assertEquals(1, dml.totalSize());
+        if (isDBS() && !isDBSMem()) {
+            assertEquals(0, dml.totalSize()); // 0 instead of -1 because DocumentModelList is weird
```

Comment:
👍  but what do you mean by "weird"

Reply:
Usually methods use `-1` as the convention for "unknown". `DocumentModelList` uses `0` for no good reason (although it predates the method in `PartialList`), which is confusing because `0` is actually a valid size.

Reply:
ah ok I understand better, thanks. "weird" indeed.
I would suggest to make the comment a little bit clearer then, but up to you!